apps/api/src/app/integrations/usecases/deactivate-integration/deactivate-integration.usecase.ts

```diff
@@ -1,11 +1,24 @@
 import { Injectable } from '@nestjs/common';
 import { IntegrationRepository } from '@novu/dal';
+import { FeatureFlagCommand, GetFeatureFlag } from '../../../feature-flags/use-cases';
 import { DeactivateSimilarChannelIntegrationsCommand } from './deactivate-integration.command';
 
 @Injectable()
 export class DeactivateSimilarChannelIntegrations {
-  constructor(private integrationRepository: IntegrationRepository) {}
+  constructor(private integrationRepository: IntegrationRepository, private getFeatureFlag: GetFeatureFlag) {}
   async execute(command: DeactivateSimilarChannelIntegrationsCommand): Promise<void> {
+    const shouldKeepActive = await this.getFeatureFlag.isMultiProviderConfigurationEnabled(
```

Comment:
Yes, do you have a better naming suggestion?

Reply:
With what is shared maybe just reinforce the idea this is because of the integration so `shouldKeepIntegrationActive` or even `skipIntegrationDeactivation`. If those suggestions bring you a better one, go for it. 🙂